### DISCUSSION TITLE
Schedule isn't respected

### DISCUSSION THREAD

**Discussion by jgoux** [renovatebot/renovate#8642] [Request Help] [answered]
Hello all :)

I've setup scheduling for renovate in our repository like this : 

```json
{
  "schedule": [
    "after 9am every weekday",
    "before 6pm every weekday"
  ],
  "timezone": "Europe/Paris"
}
```

But we still receive PR from Renovate outside of these time frames, is my schedule declaration wrong?

We want PRs to be opened between 9am and 6pm from monday to friday. 🕐 

Thanks for your help!

**Comment by viceice** [comment]
You need to share some more details, as this should work.

See here for working sample
https://github.com/renovatebot/renovate/blob/master/lib/config/presets/internal/schedule.ts#L36-L44

**Reply by jgoux** [reply]
For example yesterday night around 11pm UTC I received this PR : 
- Update dependency typescript to v4.1.5

![image](https://user-images.githubusercontent.com/1443499/107614878-71a64080-6c4b-11eb-8c17-46bb8ee8441e.png)

Which other details should I provide?

**Reply by rarkins** [reply]
1. What's your full repo config?
2. Log into app.renovatebot.com and locate the job log for when that PR was created (use timestamps to correlate). There will be logs about "schedule" if you search

**Reply by viceice** [reply]
I know now, you've misinterpreted the schedules.

Multiple schedules are combined with `or` so first matching schedule passes. try:

```json
{
  "schedule": [
    "9am through 6pm every weekday",
  ],
  "timezone": "Europe/Paris"
}
```

we are using this to parse schedules: https://breejs.github.io/later/parsers.html#text

**Reply by jgoux** [reply]
I think that's it @viceice !

I was confused by this in your documentation : 
![image](https://user-images.githubusercontent.com/1443499/107616329-204b8080-6c4e-11eb-839b-fc664c82c5ff.png)

Is it a mistake?

Thanks both of you, as always, the support is so fast! 👏

**Reply by viceice** [reply]
> I was confused by this in your documentation :
> ![image](https://user-images.githubusercontent.com/1443499/107616329-204b8080-6c4e-11eb-839b-fc664c82c5ff.png)
> 
> Is it a mistake?
> 

Nope, because those schedules do not overlap, so they need to be `or`-ed 🙃 

Read it as `after 10pm every weekday or before 5am every weekday or every weekend` 


so your first schedule was like `after 9am every weekday or before 6pm every weekday` so even 11am matches 😉

**Reply by jgoux** [reply]
Oh ok got it, I have to think in term of "from midnight to 5am" and "from 10pm to midnight"!

**Reply by jgoux** [reply]
> I know now, you've misinterpreted the schedules.
> 
> Multiple schedules are combined with `or` so first matching schedule passes. try:
> 
> ```json
> {
>   "schedule": [
>     "9am through 6pm every weekday",
>   ],
>   "timezone": "Europe/Paris"
> }
> ```
> 
> we are using this to parse schedules: https://breejs.github.io/later/parsers.html#text

By the way, the proposed syntax doesn't work (parse error), the right one is : 

```
"after 9am and before 6pm every weekday"
```

👍

**Comment by viceice** [comment, marked as answer]
Solution:
```json
{
  "schedule": [
    "after 9am and before 6pm every weekday",
  ],
  "timezone": "Europe/Paris"
}
```
